### ISSUE TITLE
Train/Val sets and respective role during optimization

### ISSUE THREAD

**Issue by bonorico** [ultralytics/yolov5#1147]
## ❔Question
Dear all,
while specifying paths to train and val data, I assume that mAP calculations (shown at the end of each iteration) are made on val data.
If this is correct, my question is:
1) is val data only used to monitor validation error during training ? (no influence on optimization)
2) Or val data also drives optimization in that validation error enters the training loss ? (influence on optimization)

While awaiting for more technical details in your upcoming report, I would be grateful, If you could already provide indications on this issue.
Thank you and best regards

**Comment by glenn-jocher** [comment]
https://en.wikipedia.org/wiki/Training,_validation,_and_test_sets

**Comment by bonorico** [comment]
Dear Glenn,
thank you for the link. However, this does still not immediately clarify what is the exact use of the validation data in your algorithm during training (if any). 
Are you for instance using a validation loss along a training loss in SGD, or are you simply using validation loss to monitor/record the validation error after each epoch in order to back-call and save the model with the "best" (vs last) weights at the end ? 
I'm really just guessing here .....
While awaiting for more documentation on the software, any hint would be greatly appreciated.
Thank you !

**Comment by glenn-jocher** [comment]
@bonorico there's nothing out of the ordinary here in terms of dataset splitting or usage. train losses are applied to optimizer updates, and train and val losses, along with val metrics are shown clearly in tensorboard. Suggest following the tutorials to get started.
https://docs.ultralytics.com/yolov5

**Comment by bonorico** [comment]
Dear Glenn,
Ok. Many thanks for you clarification ;)

**Comment by Alex-afka** [comment]
> 亲爱的格伦，
> 好的。非常感谢您的澄清;）

is val no influence on optimization？i have the same question

**Comment by glenn-jocher** [comment]
@Alex-afka there's nothing out of the ordinary here in terms of dataset splitting or usage. train losses are applied to optimizer updates, and train and val losses, along with val metrics are shown clearly in tensorboard. Suggest following the tutorials to get started.
https://docs.ultralytics.com/yolov5
